paddle/memory/memcpy.cc

```diff
@@ -29,6 +29,16 @@ void Copy<platform::CPUPlace, platform::CPUPlace>(platform::CPUPlace, void* dst,
 }
 
 #ifndef PADDLE_ONLY_CPU
+
+template <>
+void Copy<platform::CPUPlace, platform::GPUPlace>(platform::CPUPlace dst_place,
+                                                  void* dst,
+                                                  platform::GPUPlace src_place,
+                                                  const void* src, size_t num) {
+  platform::SetDeviceId(src_place.device);
+  platform::GpuMemcpySync(dst, src, num, cudaMemcpyDeviceToHost);
```

Comment:
It is very strange that if we invoke some copy method in memory.h, it will trigger link error while compiling.

It is hard to debug if the developer is not familiar with C++, template, and memory.{h/cc}.

So, we should implement the Copy correctly in memory.{h/cc}. It is developer's choice to add a stream or not.